aibolit/metrics/NumberMethods/NumberMethods.py

```diff
@@ -0,0 +1,35 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2020 Aibolit
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from aibolit.ast_framework import AST, ASTNodeType
+from aibolit.utils.ast_builder import build_ast
+
+
+class NumberMethods:
+    def value(self, filename: str) -> int:
+        metric = 0
+        ast = AST.build_from_javalang(build_ast(filename))
+        for class_node in ast.get_proxy_nodes(ASTNodeType.CLASS_DECLARATION):
+            tree = ast.get_subtree(class_node)
+            for method_node in tree.get_proxy_nodes(ASTNodeType.METHOD_DECLARATION):
```

Comment:
You can use `class_node.methods` field to iterate over all methods automatically. In current implementation we will also count all methods of nested classes, which does not seem right.

Reply:
ok